### DISCUSSION TITLE
Maximum number of consumers I.e. active application instance per quorum queue

### DISCUSSION THREAD

**Discussion by msschl** [rabbitmq/rabbitmq-server#11504] [Questions]
What is the maximum number of consumers I.e. active application instance per quorum queue?

We receive millions of raw frames 868mhz from IOT devices that we enqueue as fast as possible, to then later consume. The consumer takes a raw frame and decodes it, stores the data to the database and acknowledges the message. To scale the decoding part we just scale up or down depending on the volume in the queue the number of consumers. The order of messages decoding is not important for us. Is there a max number of consumers that a single quorum queue should have? Or is it better practice to have a dedicated quorum queue per consumer instance and spread the load across the quorum queues ?

**Comment by msschl** [comment]
@michaelklishin

**Comment by kjnilsson** [comment]
The best way is to simulate different scenarios using https://perftest.rabbitmq.com/

How many consumers that are optimal for a given workload isn't easy to say but at some point having too many results in diminishing returns. I don't think having 1000s of consumers is a good idea but 100 could be for some types of processing. Prefetch sizes also play a part.
